src/launchbox/token/LaunchboxFactory.sol

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/proxy/Clones.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import {LaunchboxERC20} from "./LaunchboxERC20.sol";
+
+contract LaunchboxFactory is Ownable(msg.sender) {
+    event TokenDeployed(address tokenAddress, address launchboxExchangeAddress);
+    event LaunchboxExchangeImplementationUpdated(address newLaunchboxExchangeAddress);
+    event TokenImplementationUpdated(address newTokenImplementationAddress);
+    event UniswapRouterUpdated(address newRouter);
+
+    address public tokenImplementation;
+    address public launchboxExchangeImplementation;
+    address public uniswapRouter;
+    
+    error EmptyTokenImplementation();
+    error EmptyLaunchboxExchangeImplementation();
+    error EmptyUniswapRouter();
+
+    constructor(address _tokenImplementation, address _launchboxExchangeImplementation, address _uniswapRouter) {
```

Comment:
We also need to implement first buy. so after the contracts gets created, the deployer should be a certain portion of the token.

Reply:
cool, but we shouldn't make it compulsory right?

Reply:
yeah

Reply:
I think we should take it out entirely. extra complexity